src/drivers/intel/cavs/interrupt.c

```diff
@@ -173,7 +173,7 @@ void interrupt_unmask(uint32_t irq)
 
 static void irq_mask(struct irq_desc *desc, uint32_t irq)
 {
-	int core = SOF_IRQ_CPU(irq);
+	int core = cpu_get_id();
```

Comment:
In function `void idc_enable_interrupts(int target_core, int source_core)` master core enables IDC interrupt for the targeted slave core.

Reply:
@tlauda good catch, thanks. BTW, looking at it, the arch_cpu_enable_core() function already allocated RAM for other cores, doesn't it? So, this patch set wouldn't be the first such precedent. As for the interrupt, wouldn't it be possible to enable that IDC interrupt in `slave_core_init()` instead?

Reply:
@lyakh It allocates memory for other cores for core context struct, which is needed for core to even start doing anything, so this is something we cannot go around and do differently.
IDC interrupt needs to be enabled for slave core, so it could receive the message to jump from ROM to FW.

Reply:
It is good that we have per-core heaps, they are useful for run-time memory allocations, avoiding inter-core locking and cache synchronisation issues. But I also don't see a problem allocating memory on the boot core for other cores. The boot core does anyway perform more work especially during initialisation. I'd propose to increase the system memory heap on the boot core, maybe we can also reduce the heap size on non-boot cores.

Reply:
@lyakh But what is the motivation to change that? Master core already has bigger heaps than slave cores. I'm a big fan of improvements and new features, but I simply don't see any benefits in your proposal. Why should the master core waste cycles and memory for doing stuff that it doesn't need to do and that probably won't be used for 90% of the time.

Reply:
@tlauda see above, let's continue the discussion in a single comment thread.